homeassistant/components/tankerkoenig/config_flow.py

```diff
@@ -0,0 +1,224 @@
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LOCATION,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_RADIUS,
+    CONF_SHOW_ON_MAP,
+    CONF_UNIT_OF_MEASUREMENT,
+    LENGTH_KILOMETERS,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import selector
+
+from .const import CONF_FUEL_TYPES, CONF_STATIONS, DEFAULT_RADIUS, DOMAIN, FUEL_TYPES
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Init the FlowHandler."""
+        super().__init__()
+        self._data: dict[str, Any] = {}
+        self._stations: dict[str, str] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import YAML configuration."""
+        await self.async_set_unique_id(
+            f"{config[CONF_LOCATION][CONF_LATITUDE]}_{config[CONF_LOCATION][CONF_LONGITUDE]}"
+        )
+        self._abort_if_unique_id_configured()
+
+        selected_station_ids: list[str] = []
+        # add all nearby stations
+        nearby_stations = await self._get_nearby_stations(config)
+        for station in nearby_stations.get("stations", []):
+            selected_station_ids.append(station["id"])
+
+        # add all manual added stations
+        for station_id in config[CONF_STATIONS]:
+            selected_station_ids.append(station_id)
+
+        return self._create_entry(
+            data={
+                CONF_NAME: "Home",
+                CONF_API_KEY: config[CONF_API_KEY],
+                CONF_FUEL_TYPES: config[CONF_FUEL_TYPES],
+                CONF_LOCATION: config[CONF_LOCATION],
+                CONF_RADIUS: config[CONF_RADIUS],
+                CONF_STATIONS: selected_station_ids,
+            },
+            options={
+                CONF_SHOW_ON_MAP: config[CONF_SHOW_ON_MAP],
+            },
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        if not user_input:
+            return self._show_form_user()
+
+        await self.async_set_unique_id(
+            f"{user_input[CONF_LOCATION][CONF_LATITUDE]}_{user_input[CONF_LOCATION][CONF_LONGITUDE]}"
+        )
+        self._abort_if_unique_id_configured()
+
+        data = await self._get_nearby_stations(user_input)
+        if not data.get("ok"):
+            return self._show_form_user(
+                user_input, errors={CONF_API_KEY: "invalid_auth"}
+            )
+        if stations := data.get("stations"):
```

Comment:
We can invert this check and return the form early if true.

Reply:
tbh I'm not sure what is the advantage of this? here the user form ist just shown in case of error - so commonly should not be shown at this point anymore 🤔

Reply:
It's just that flat is better than nested, I think.

Reply:
ah ... now I get it